### PR TITLE
Campaign scenario UI: show the currently selected bonus if the campaign scenario is already in progress

### LINKED ISSUE #7274: Info shows wrong choice in campaign
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

Start a campaign, and you'll be presented with 3 choices. Select second or third choice, and start the map. Then click Info button. It will claim that you selected the first choice.

Рабочий стол 1_365

I chose barb here, not necro

### Save file

[Evil_4_0026.savc.gz](https://github.com/ihhub/fheroes2/files/11654879/Evil_4_0026.savc.gz)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #7274

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @oleg-derevenetz , I think we need to change the way how we store campaign data. Yes, this change works fine but in the future we need to fix the problem with browsing previous completed scenarios and there we will have problem as we don't save any of them. The best approach would be to store bonus ID for each scenario instead of currently selected bonus.

What do you think?

**Comment by oleg-derevenetz** [comment]
@ihhub are we able to browse the previously completed scenarios using the original UI? I thought that there are auto-saves for this at the completion of each scenario. I propose to implement this as soon as we directly approach the implementation of viewing the campaign scenario history, and do not engage in overengineering in advance, because it is not a fact that some kind of implementation "from the head" will then fit.

**Comment by ihhub** [comment]
@oleg-derevenetz , yes, it is possible to browse previously completed scenarios in the original game.

**Comment by oleg-derevenetz** [comment]
@ihhub ah, OK then, I'll try to rework the bonus storage in this PR later.

File: src/fheroes2/campaign/campaign_savedata.cpp [hunk 1]
```diff
@@ -153,14 +160,42 @@ namespace Campaign
...
+        static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1005_RELEASE, "Remove the logic below." );
+        if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1005_RELEASE ) {
+            // There is always a bonus in all the original HoMM2 campaign missions
+            data._bonusesForFinishedMaps.resize( data._finishedMaps.size(), 0 );
+        }
+        else {
+            msg >> data._bonusesForFinishedMaps;
+
+            // Make sure that the number of elements in the vector of map bonuses matches the number of elements in the vector of finished maps
+            data._bonusesForFinishedMaps.resize( data._finishedMaps.size(), -1 );
```
Comment (ihhub) [inline src/fheroes2/campaign/campaign_savedata.cpp:195]: Could you please explain why we are setting -1 here instead of 0?

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left one comment here. Would you mind to take a look?

File: src/fheroes2/campaign/campaign_savedata.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/campaign/campaign_savedata.cpp:195]: Hi @ihhub We can set 0 as well, but:

1. Normally this should be a no-op;
2. If this is not a no-op, then I prefer to emphasize here that there is no information about selected bonus, and value 0 means that the first bonus was selected. Also in future we May have a campaign where there will be no bonuses for certain missions.

Comment (ihhub) [reply src/fheroes2/campaign/campaign_savedata.cpp:195]: Good points. Thanks for the explanation!

**Comment by ihhub** [comment]
@oleg-derevenetz , huge thanks for these improvements!

**Pull request merged**